src/Content/OEmbed.php

```diff
@@ -184,13 +184,13 @@ private static function fetchURL(string $embedurl): \Friendica\Object\OEmbed
 
 		$eventDispatcher = DI::eventDispatcher();
 
-		$oembed_data = ['url' => $embedurl];
+		$oembed_data = ['data' => $oembed];
 
 		$oembed_data = $eventDispatcher->dispatch(
 			new ArrayFilterEvent(ArrayFilterEvent::OEMBED_FETCH_END, $oembed_data),
 		)->getArray();
 
-		return $oembed_data['url'] ?? $embedurl;
+		return $oembed_data['data'];
```

Comment:
```suggestion
		return $oembed;
```

Reply:
I think that an addon should be able to alter the oembed data.

Reply:
I agree but for BC reasons this hook only supports an url, see https://github.com/friendica/friendica/blob/82ea62ba7cba358174a54de6f7ef341e4137263a/src/Core/Hooks/HookEventBridge.php#L137-L146

We cannot replace the url with the oembed object because atm the listening addons expecting a string. Once we have refactored everything to the EventDispatcher, we can deprecate the hooks and switch to the Event objects. Lets fix the return object in this PR and move into a separate PR for the addition with the oembed object.

Reply:
Is there any addon that uses this? The addon only makes sense when it uses the already processed object and returns it.

Reply:
No official addons uses this.

Reply:
I also think that this hook can't be used for anything useful at the moment, since it doesn't have got access to the OEmbed data but just the URL. 

As we can see in the previous hook call, the data was intended to be passed to the hook: `Hook::callAll('oembed_fetch_url', $embedurl, $oembed);`. Only thing was, that wasn't implemented correctly. So we now should fix it (but not by just removing that parameter).

Reply:
As I mentiond we can provide the oembed object in addition. 

```php
		$oembed_data = [
			'url' => $embedurl,
			'oembed' => $oembed,
		];
```

This will not fix the current hook but will be available once we switched to the EventDispatcher events.